api/server.go

```diff
@@ -48,28 +50,58 @@ func NewServer(config common.Config) error {
 		return errors.New("'org' cannot be empty in the configuration")
 	}
 	Org = config.Org
-	repo := config.Repository
+	metadata := config.MetadataRepository
 
-	// Create metadata repository session
-	log.Debugf("Creating new MetadataRepository of type %s with configuration %+v (org: %s)", repo.Type, repo.Config, Org)
+	// Initialize metadata repository session
+	log.Debugf("Creating new session for MetadataRepository of type %s with configuration %+v (org: %s)", metadata.Type, metadata.Config, Org)
 
 	var metadataRepo dataset.MetadataRepository
 	var err error
-	switch repo.Type {
+	switch metadata.Type {
 	case "s3":
-		metadataRepo, err = s3metadatarepository.NewDefaultRepository(repo.Config)
+		if metadata.Config["prefix"] == nil {
+			metadata.Config["prefix"] = Org
+		} else {
+			metadata.Config["prefix"] = metadata.Config["prefix"].(string) + "/" + Org
```

Comment:
this will panic if it's not a string - I don't think you want that.  thats why i do all of my assertions in the NewXxxx method.  

You could do something like...

```golang
prefix := Org
if c, ok := metadata.Config["prefix"]; ok {
  if p, ok := c.(string); ok {
    prefix = p + "/" + prefix
  }
}
metadata.Config["prefix"] = prefix
```